process/block/baseProcess.go

```diff
@@ -73,11 +72,10 @@ func (bp *baseProcessor) RevertAccountState() {
 	}
 }
 
-// SetLastNotarizedHdr sets the last notarized header
-func (bp *baseProcessor) SetLastNotarizedHdr(shardId uint32, processedHdr data.HeaderHandler) {
+// AddLastNotarizedHdr adds the last notarized header
+func (bp *baseProcessor) AddLastNotarizedHdr(shardId uint32, processedHdr data.HeaderHandler) {
 	bp.mutNotarizedHdrs.Lock()
-	bp.finalNotarizedHdrs[shardId] = bp.lastNotarizedHdrs[shardId]
-	bp.lastNotarizedHdrs[shardId] = processedHdr
+	bp.lastNotarizedHdrs[shardId] = append(bp.lastNotarizedHdrs[shardId], processedHdr)
```

Comment:
Actually this is a list which hold the history of notarized headers in the other chain at the moment of each new commit of the new header of its own chain. It would often happen that this to have the same header added many times one after one. If we take the case when let's say a node from shard 0 commits block with nonce 10 and let's say the highest meta block nonce used to validate this shard block has nonce 8 at the time of  the commit, it will append at this notarized list the block with nonce 8. If for the next two headers committed (11,12) nothing new will be used from meta chain, than would be another two appends with the same header with nonce 8. So the list will keep 3 nonces with value of 8. If the node will do a roll back because of any reason (fork detection for example), and it will go behind with let's say two blocks, than it will have the correct metachain notarized nonce set to 8 after it will remove the last two items from this list.

Reply:
Ok, got it.